### PR TITLE
feat(azure): new check related to vulnerability assessment `sqlserver_vulnerability_assessment_enabled`

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

This pr contains new check: `sqlserver_vulnerability_assessment_enabled`

### Description

This check makes the task:

- Ensure that Vulnerability Assessment (VA) is enabled on a SQL server by setting a Storage Account

*Check Logic:* When vulnerability assessment is not none and the storage_container_path has a value, it has a vulnerability assessment so the check will pass

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "Description": "Enable Vulnerability Assessment (VA) service scans for critical SQL servers and corresponding SQL databases.",
+  "Risk": "The Vulnerability Assessment service scans databases for known security vulnerabilities and highlights deviations from best practices, such as misconfigurations, excessive permissions, and unprotected sensitive data. Results of the scan include actionable steps to resolve each issue and provide customized remediation scripts where applicable. Additionally, an assessment report can be customized by setting an acceptable baseline for permission configurations, feature configurations, and database settings.",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment",
+  "Remediation": {
+    "Code": {
+      "CLI": "Update-AzSqlServerVulnerabilityAssessmentSetting `-ResourceGroupName 'resource group name'` -ServerName 'Server Name'`-StorageAccountName 'Storage Name from same subscription and same Location' `-ScanResultsContainerName 'vulnerability-assessment' `-RecurringScansInterval Weekly `-EmailSubscriptionAdmins $true `-NotificationEmail @('mail1@mail.com' , 'mail2@mail.com')",
```
Comment (jfagoagas) [inline prowler/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled.metadata.json]:
```suggestion
      "CLI": "Update-AzSqlServerVulnerabilityAssessmentSetting -ResourceGroupName <resource group name> -ServerName <Server Name> -StorageAccountName <Storage Name from same subscription and same Location> -ScanResultsContainerName <vulnerability-assessment> -RecurringScansInterval Weekly -EmailSubscriptionAdmins $true -NotificationEmail @('mail1@mail.com' , 'mail2@mail.com')",
```

Comment (jfagoagas) [reply prowler/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled.metadata.json]: Please review the syntax of the above command.

File: tests/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled_test.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    ServerVulnerabilityAssessment,
+    TransparentDataEncryption,
+)
+
+from prowler.providers.azure.services.sqlserver.sqlserver_service import (
+    DatabaseServer,
+    SQL_Server,
+)
+
+AZURE_SUSCRIPTION = str(uuid4())
```
Comment (jfagoagas) [inline tests/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled_test.py:15]: Remember that this value is present in the fixtures.

**Review by jfagoagas — APPROVED** [review]
Great job @pedrooot 🚀 

Please review both comments to merge it!

**Review by jfagoagas — CHANGES_REQUESTED** [review]
I set the wrong response.

File: tests/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled_test.py (hunk 2 above)
Comment (pedrooot) [reply tests/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled_test.py:15]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled.metadata.json (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_vulnerability_assessment_enabled/sqlserver_vulnerability_assessment_enabled.metadata.json]: Done!

**Review by jfagoagas — DISMISSED** [review]
🚀

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**